src/ardrive.ts

```diff
@@ -213,38 +244,180 @@ export class ArDrive extends ArDriveAnonymous {
 			],
 			tips: [tipData],
 			fees: {
-				[uploadFileResult.metaDataTrxId]: +uploadFileResult.metaDataTrxReward,
 				[uploadFileResult.dataTrxId]: +uploadFileResult.dataTrxReward,
+				[uploadFileResult.metaDataTrxId]: +uploadFileResult.metaDataTrxReward,
 				[tipData.txId]: +communityTipTrxReward
 			}
 		});
 	}
 
+	public async createPublicFolderAndUploadChildren(
+		parentFolderId: FolderID,
+		wrappedFolder: ArFSFolderToUpload,
+		parentFolderName?: string
+	): Promise<ArFSResult> {
+		const driveId = await this.getDriveIdAndAssertDrive(parentFolderId);
+
+		const parentFolderData = new ArFSPublicFolderTransactionData(
+			parentFolderName ?? wrappedFolder.getBaseFileName()
+		);
+
+		// Estimate and assert the cost of the entire bulk upload
+		// This will assign the calculated base costs to each wrapped file and folder
+		const bulkEstimation = await this.estimateAndAssertCostOfBulkUpload(wrappedFolder, 'public', parentFolderData);
```

Comment:
> Can we hoist total cost estimation and wallet balance assertion into a bulk spot in the ardrive layer?

@arielmelendez 

This is already the case in this PR. The estimation is used here and implemented below in the `estimateAndAssertCostOfBulkUpload` method